### PR TITLE
CATROID-1367 Add single undo functionality in datalist

### PR DESCRIPTION
**Pull request by sebastiangollob**
Added single undo functionality in datalist. Allows to undo rename, delete and edit of variables in the datalist.

https://jira.catrob.at/browse/CATROID-1367

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by bengaal — APPROVED** [review]
Looks good to me! Thank you! :)

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/adapter/DataListAdapter.java [hunk 1]
```diff
@@ -110,10 +128,25 @@ public void onBindViewHolder(CheckableViewHolder holder, int position) {
 			protected void onCheckBoxClick(int position) {
 				super.onCheckBoxClick(getRelativeItemPosition(position, VAR_GLOBAL));
 			}
+
+			@Override
+			public List<UserVariable> filterItems(List<UserVariable> all_items) {
```
Comment (mseiser) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/adapter/DataListAdapter.java:133]: Code duplication, doesn't conform to clean code guidelines, please remove duplicated code.

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/adapter/DataListAdapter.java [hunk 2]
```diff
@@ -126,6 +159,21 @@ public void onBindViewHolder(CheckableViewHolder holder, int position) {
 			protected void onCheckBoxClick(int position) {
 				super.onCheckBoxClick(getRelativeItemPosition(position, VAR_LOCAL));
 			}
+
+			@Override
+			public List<UserVariable> filterItems(List<UserVariable> all_items) {
```
Comment (mseiser) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/adapter/DataListAdapter.java:164]: Code duplication

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/DataListFragment.kt [hunk 3]
```diff
@@ -389,11 +509,57 @@ class DataListFragment : Fragment(),
...
         }
         return true
     }
 
+    private fun undoLastPerformedAction() {
+        if (lastPerformedAction != null) {
+            var alertTitle = ""
+            var alertMsg = ""
+
+            when (lastPerformedAction!!.actionType) {
```
Comment (mseiser) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/DataListFragment.kt:523]: using !! increases risk of crashing, other ways like using elvis ?: are safer, please change

**Review by mseiser — CHANGES_REQUESTED** [review]

**Review by mseiser — APPROVED** [review]

**Pull request merged**